modules/aleph/includes/aleph.reservation.inc

```diff
@@ -213,8 +215,67 @@ function aleph_reservation_list($account, $requested_type = NULL) {
   return $result;
 }
 
-function aleph_reservation_create() {
-  return array();
+/**
+ * Create a new reservation in the library system.
+ *
+ * @param object $account
+ *   Drupal user account object.
+ * @param string $id
+ *   ID of reservation.
+ *
+ * @return array
+ *   Reservation information.
+ *
+ * @throws \DingProviderReservationExists
+ *   Reservation exists.
+ * @throws \DingProviderUserException
+ *    General exception.
+ */
+function aleph_reservation_create($account, $id) {
+  $patron_handler = new AlephPatronHandler(aleph_client());
+  $patron_handler->setPatron(aleph_get_patron_by_account());
+
+  // Check if the users has this reservation and throw exception.
+  $reservations = $patron_handler->getReservations();
+
+  foreach ($reservations as $reservation) {
+    if ($id === $reservation->getItem()->getId()) {
+      throw new DingProviderReservationExists();
+    }
+  }
+
+  // Add the interest period for the 'to' date.
+  $interest_period = aleph_reservation_default_interest_period();
+
+  $end_request_date = new DateTime();
+  $end_request_date->add(new DateInterval('P' . $interest_period . 'D'));
+
+  // Build the reservation object.
+  $defaults = aleph_reservation_default_options($account);
+  $reservation = new AlephReservation();
+  $request = new AlephRequest();
+  $request->setRequestDate(new DateTime());
+  $request->setEndRequestDate($end_request_date);
+  $request->setPickupLocation($defaults['aleph_preferred_branch']);
+  $reservation->setRequest($request);
+
+  // Try to make the reservation.
+  $patron = $patron_handler->getPatron();
+  $response = NULL;
+  try {
+    $response = $patron_handler->createReservation($patron, $reservation);
+  }
+  catch (Exception $e) {
+    watchdog_exception('aleph', $e);
+  }
+
+  if ($response->success()) {
```

Comment:
If an exception occurs for `createReservation()` then `$response` will be `NULL` and you will get an error trying to call a method on a non-object.